### DISCUSSION TITLE
Hot Reloading Issue

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#17001] [Q&A] [answered]
I have a `dist/index.html` file that refers to a bundled JS script. However, hot reloading only works when this script is in the same directory as `index.html` but fails when I place the script in a subdirectory (`dist/assets`). Check out my entire project structure:

```
webpack-test
│   .babelrc
│   webpack.config.js
│   package-lock.json
│   package.json
│
├───dist
│   │   index.html
│   │   bundle.js
│   │
│   └───assets
│           bundle.js
│
└───src
        index.js
```

Here is my webpack config:

```
const path = require('path');

module.exports = {
    entry: './src/index.js',
    output: {
        path: path.resolve(__dirname, 'dist/assets'),
        filename: 'bundle.js'
    },
    devServer: {
        port: 9000,
        compress: true,
        hot: true,
        static: {
            directory: path.join(__dirname, 'dist')
        }
    }
}
```

So when I indicate `<script src="./index.js"></script>` in `dist/index.html` hot reloading works as expected. It doesn't work when I do `<script src="./assets/index.js"></script>` though. I suppose the config is correct but I'm not sure.

**Comment by alexander-akait** [comment]
Because dev server use `output.path` as a base directory to handle bundled files, you can reconfigure it

**Reply by ghost** [reply]
Thanks for pointing it out! Any chance you can tell me what property would govern this setting exactly?

**Reply by alexander-akait** [reply, marked as answer]
You can change dist to public and handle it from there, but I recommend to do nothing, also you can look at `HtmlWebpackPlugin` options
